guides/micronaut-cloud-secrets-azure/micronaut-cloud-secrets-azure.adoc

```diff
@@ -0,0 +1,194 @@
+
+https://docs.micronaut.io/latest/guide/#bootstrap[Click here to get more details on distributed configuration]
+
+=== Clean up of the Application Configuration
+
+Remove the redundant application name entry from the `application.yml`.
+
+[,diff]
+.src/main/resources/application.yml
+----
+- micronaut:
+-  application:
+-    name: micronautguide
+
+datasources:
+  default:
+    username: ${JDBC-USER}
+    password: ${JDBC-PASSWORD}
+    dialect: MYSQL
+    driverClassName: ${JDBC_DRIVER:com.mysql.cj.jdbc.Driver}
+
+flyway:
+  datasources:
+    default:
+      enabled: true
+----
+
+== Store JDBC secrets in Azure Key Vault using Azure Portal
+
+Create Azure Key Vault service resource.
+{empty} +
+
+image::azure-key-vault/azure-vault-create-vault-resouce.png[]
+{empty} +
+
+When Key Vault service deployment is ready, please go to the resource in order to create secrets. Prior to clicking on the Secrets option in the left menu list, please copy the Vault URI from the main screen section. You will need to set it as an environment variable in order to integrate Azure Key Vault into your application.
+{empty} +
+
+image::azure-key-vault/azure-vault-obtain-vault-uri.png[]
+{empty} +
+
+Create new secrets for JDBC user and password. Click on the Secrets option in the left menu, then click on Generate/Import.
+{empty} +
+
+image::azure-key-vault/azure-vault-create-secret-jdbc-user.png[]
+{empty} +
+
+image::azure-key-vault/azure-vault-create-secret-jdbc-password.png[]
+{empty} +
+
+Go to App registration in order to register your application with the Microsoft identity platform. After registering your application,
+copy the secret Application (client) ID, and Directory (tenant) ID. You will use them later on as environment variables in order to connect to Azure Key Vault.
+{empty} +
+
+image::azure-key-vault/azure-register-application.png[]
+{empty} +
+
+Navigate to your application profile in order to generate application credentials that you will use to authenticate with the Azure Key Vault service. In the left menu, click on the Certificates & secrets. To generate the application credentials, click on the Client secrets tab and then on the New client secret button. Copy the secret value. You will need to set it along the Application (client) ID, Directory (tenant) ID, and Vault URI as environment variables.
+NOTE: The client's secret value cannot be viewed, except immediately after creation. Be sure to save the secret when created before leaving the page
+{empty} +
+
+image::azure-key-vault/azure-register-application-add-client-secret.png[]
+{empty} +
+
+In order to finish the Key Vault setup process. you will need to assign a proper policy to the Key Vault resource. Go to your Key Vault resource profile. Click on the Access policies option in the left menu. Finally, create an access policy.
+{empty} +
+
+image::azure-key-vault/azure-vault-create-policy.png[]
```

Comment:
the policy should define the minimum amount of access. Is not just "List" and "Get" enough?

Reply:
Yes, I think you are right. I will make the changes.

Reply:
Covered by https://github.com/micronaut-projects/micronaut-guides/pull/1099/commits/8fed9a79fe5adb0859856581038fe14bef237f02.